### ISSUE TITLE
missing broadcast address in facts

### ISSUE THREAD

**Issue by neutrinus** [ansible/ansible#10917]
I try to template the /etc/network/interfaces in debian, need broadcast address. Unfortunetly, there is no broadcast address in ansible facts, part of setup result:

```
     "ansible_default_ipv4": {
            "address": "144.76.19.151", 
            "alias": "eth0", 
            "gateway": "144.76.19.129", 
            "interface": "eth0", 
            "macaddress": "4c:72:b9:7c:c6:1d", 
            "mtu": 1500, 
            "netmask": "255.255.255.224", 
            "network": "144.76.19.128", 
            "type": "ether"
        }, 
       "ansible_eth0": {
            "active": true, 
            "device": "eth0", 
            "ipv4": {
                "address": "144.76.19.151", 
                "netmask": "255.255.255.224", 
                "network": "144.76.19.128"
            }, 
            "ipv6": [
                {
                    "address": "2a01:4f8:191:1196::2", 
                    "prefix": "64", 
                    "scope": "global"
                }, 
                {
                    "address": "fe80::4e72:b9ff:fe7c:c61d", 
                    "prefix": "64", 
                    "scope": "link"
                }
            ], 
            "macaddress": "4c:72:b9:7c:c6:1d", 
            "module": "e1000e", 
            "mtu": 1500, 
            "promisc": false, 
            "type": "ether"
        }, 
```

_ansible 1.9.1_, debian jessie. 

I see that previously this was working (https://github.com/ansible/ansible/issues/8519), can we have it back? I found even similar question (http://grokbase.com/t/gg/ansible-project/14c2jdv7qn/ipv4-broadcast-address) but without answers..

**Comment by mightydok** [comment]
@neutrinus i fix error, you can manualy apply 2 commits attached to this issue, or wait for this pull request: https://github.com/ansible/ansible/pull/10971

**Comment by giviz** [comment]
Seems that the problem is back on debian 8 with ansible 1.9.4, facts does not provide broadcast address anymore.
facts.py seems to have changed a lot since last may, does the error back ?

**Comment by bcoca** [comment]
@giviz never left, fix is in >= 2.0

**Comment by giviz** [comment]
Oh, my bad :)
Will add the value to host_vars until we migrate to v2 then !
Thanks
